packages/core/content-manager/server/controllers/relations.js

```diff
@@ -18,6 +18,29 @@ const addFiltersClause = (params, filtersClause) => {
   }
 };
 
+const MAIN_FIELD_READ_ALLOWED_MODELS = ['plugin::users-permissions.role'];
```

Comment:
@innerdvations this seems a bit vague and wondering if we should limit this more, saying we allow to read the role main field if the model that contains this relation is `plugin::users-permissions.user`

Reply:
Yes, I would do it like this:

Inside the permissionChecker.cannot.read block below (so it only happens if the check fails), check if this is the role field.

If it is the role field && the user has permission for reading users-permissions.user, return 'name' (not mainfield). Otherwise it falls through and returns 'id' as normal

Reply:
updated! let me know what you think

Reply:
This looks like a good approach and is working for me! feel free to ping me again once the tests are ready